skfp/preprocessing/filters/lipinski.py

```diff
@@ -0,0 +1,97 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumLipinskiHBA, CalcNumLipinskiHBD
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class LipinskiFilter(BaseFilter):
+    """
+    Lipinski's Rule of 5 (RO5).
+
+    Also known as Pfizer's Rule of 5. It evaluates the drug-likeness of a molecule
+    as an orally active drug. Assumes that it should be small and lipophilic.
+
+    Molecule can violate at most one of the rules (conditions):
+    - molecular weight <= 500 daltons
+    - hydrogen bond acceptors (HBA) <= 10
+    - hydrogen bond donors (HBD) <= 5
+    - octanol-water partition coefficient (logP) <= 5
+
+    HBA and HBD use a simplified definition, taking into consideration only oxygen
+    and nitrogen bonds with hydrogen (OH, NH).
+
+    Parameters
+    ----------
+    allow_one_violation : bool, default=True
+        Whether to allow violating one of the rules for a molecule. This makes the
+        filter less restrictive, and is the part of the original definition of this
+        filter.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See Scikit-learn documentation on ``n_jobs`` for more details.
+
+    batch_size : int, default=None
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    ----------
+    .. [1] `Christopher A. Lipinski, Franco Lombardo, Beryl W. Dominy, Paul J. Feeney
+        "Experimental and computational approaches to estimate solubility and permeability
+        in drug discovery and development settings"
+        Advanced Drug Delivery Reviews, Volume 23, Issues 1–3, 15 January 1997, Pages 3-25
+        <https://www.sciencedirect.com/science/article/pii/S0169409X96004231>`_
+
+    Examples
+    --------
+    >>> from skfp.preprocessing import LipinskiFilter, MolToSmilesTransformer
+    >>> smiles = ["[C-]#N", "CC=O", "O=C(O)c1ccccc1c2ccc(cc2)Cn3c4cc(cc(c4nc3CCC)C)c5nc6ccccc6n5C"]
+    >>> filt = LipinskiFilter()
+    >>> filt
+    LipinskiFilter()
+
+    >>> filtered_mols = filt.transform(smiles)
+    >>> mol_to_smiles = MolToSmilesTransformer()
```

Comment:
There's no need for MolToSmilesTransformer -  filter returns SMILES anyway.

Reply:
Right, I forgot about that